src/datalink/mod.rs

```diff
@@ -119,21 +119,21 @@ impl DataLinkReceiver {
     #[inline]
     pub fn iter<'a>(&'a mut self) -> DataLinkChannelIterator<'a> {
         DataLinkChannelIterator {
-            imp: self.dlri.iter()
+            channel_iterator: self.receiver.iter()
         }
     }
 }
 
 /// An iterator over data link layer packets
 pub struct DataLinkChannelIterator<'a> {
-    imp: backend::DataLinkChannelIteratorImpl<'a>,
+    channel_iterator: backend::DataLinkChannelIteratorImpl<'a>,
```

Comment:
Again, I think this would be better named `backend` - it's a little confusing to have a `channel_iterator` inside a `channel_iterator`. Perhaps it would be worth looking at what other projects call this member, I know other projects use similar techniques.